python_modules/dagster/dagster/_core/definitions/external_asset_graph.py

```diff
@@ -108,7 +108,9 @@ def from_repository_handles_and_external_asset_nodes(
             if not node.is_source
         }
         job_names_by_key = {
-            node.asset_key: node.job_names for _, node in repo_handle_external_asset_nodes
+            node.asset_key: node.job_names
+            for _, node in repo_handle_external_asset_nodes
+            if not node.is_source
```

Comment:
IIRC this is reversing a change I made that was part of allowing observable source assets to run in asset jobs, which is probably why the stale scheduler test is failing.